### PR TITLE
Implement Subheadline component

### PR DESCRIPTION
**Pull request by clxandstuff**
New component to easily use new Untitled Serif font on brainly marketing pages.

Simple example:

```jsx
<Subheadline>sub header</Subheadline>
```

Advanced example:

```jsx
<Subheadline inherited type="span" size="xsmall" color="gray">
    nested subheadline with inherited styles
</Subheadline>
```

All props:

- **children** - content
- **size** - one of font sizes
- **type** - HTML tag to render
- **color** - text color
- **transform** - text transforms e.g. uppercase
- **align** - text horizontal alignment
- **className** - class names to attach
- **inherited** - to inherit styles from parent components

### CODE REVIEW

**Comment by coderitual** [comment]
@clxandstuff how about chromatic?

**Comment by clxandstuff** [comment]
I will add it tomorrow along with improved script for generating chromatic stories 🙂

File: src/components/text/Subheading.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  | 'text-black'
+  | 'text-white'
+  | 'text-gray-70'
+  | 'text-gray-60'
+  | 'text-gray-40'
+  | 'text-green-60'
+  | 'text-red-60'
+  | 'text-indigo-60'
+  | 'text-yellow-60'
+  | 'tetx-blue-60';
```
Comment (domidomi) [inline src/components/text/Subheading.jsx:35]: Typo

Comment (clxandstuff) [reply src/components/text/Subheading.jsx:35]: Nice catch, thanks 😄

**Comment by coderitual** [comment]
@clxandstuff looks great! Could you please add text tokens for tailwind:
- ‘serif’ font in https://github.com/brainly/style-guide/blob/master/src/css/config/fontFamily.js
- size and line height in https://github.com/brainly/style-guide/blob/master/src/css/config/fontSize.js

File: src/components/text/Subheading.jsx (hunk 1 above)
Comment (clxandstuff) [reply src/components/text/Subheading.jsx]: Fixed

**Comment by clxandstuff** [comment]
@coderitual 
I didn't know that, thanks. I added font and found missing font size for Headline 😃

**Review by coderitual — APPROVED** [review]

**Review by domidomi — CHANGES_REQUESTED** [review]

File: src/components/text/Subheading.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+export type SubheadingSizeType =
+  | 'xsmall'
+  | 'small'
+  | 'medium'
+  | 'large'
+  | 'xlarge'
+  | 'xxlarge'
+  | 'xxxlarge';
+
+export type SubheadingColorType =
```
[the remark this answers is not in the document]
Comment (clxandstuff) [reply src/components/text/Subheading.jsx]: That makes sense, I updated Subheading based on other text components.

File: src/components/text/_subheadings.scss [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    &--xxlarge {
+      @include subheadingTypeSizeVariant(xxlarge);
+    }
+
+    &--xxxlarge {
+      @include subheadingTypeSizeVariant(xxxlarge);
+    }
+
```
Comment (domidomi) [inline src/components/text/_subheadings.scss:86]: You can use textColor() mixin here

Comment (clxandstuff) [reply src/components/text/_subheadings.scss:86]: ah its already available 😄

File: src/components/text/Subheading.jsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  transform?: ?SubheadingTransformType,
+  align?: ?SubheadingAlignType,
+  className?: ?string,
+  inherited?: ?boolean,
+  ...
+};
+
+const Subheading = ({
+  children,
+  type = SUBHEADING_TYPE.H1,
```
Comment (katarzynatobis) [inline src/components/text/Subheading.jsx:57]: This is not safe from accessibility perspective

File: src/components/text/_subheadings.scss (hunk 3 above)
Comment (clxandstuff) [reply src/components/text/_subheadings.scss:88]: Done :)

File: src/components/text/Subheading.jsx (hunk 4 above)
Comment (coderitual) [reply src/components/text/Subheading.jsx:57]: I agree, h2 will be better as default. 

This applies to all other text components (out of scope of this pr).

Comment (clxandstuff) [reply src/components/text/Subheading.jsx:57]: Ok, I'm changing it to h2.

Comment (clxandstuff) [reply src/components/text/Subheading.jsx]: Done

**Review by domidomi — APPROVED** [review]

**Pull request merged**